scaler/handlers.go

```diff
@@ -38,18 +38,39 @@ func (e *impl) IsActive(
 	ctx context.Context,
 	scaledObject *externalscaler.ScaledObjectRef,
 ) (*externalscaler.IsActiveResponse, error) {
+	lggr := e.lggr.WithName("IsActive")
+	host, ok := scaledObject.ScalerMetadata["host"]
+	if !ok {
+		err := fmt.Errorf("no 'host' field found in ScaledObject metadata")
+		lggr.Error(err, "ScaledObjectRef", scaledObject)
+		return nil, err
+	}
+	if host == "interceptor" {
+		return &externalscaler.IsActiveResponse{
+			Result: true,
+		}, nil
+	}
+	allCounts := e.pinger.counts()
+	hostCount, ok := allCounts[host]
+	if !ok {
+		err := fmt.Errorf("host '%s' not found in counts", host)
+		lggr.Error(err, "allCounts", allCounts)
+		return nil, err
+	}
+	var active bool = hostCount > 0
 	return &externalscaler.IsActiveResponse{
-		Result: true,
+		Result: active,
 	}, nil
 }
 
 func (e *impl) StreamIsActive(
-	in *externalscaler.ScaledObjectRef,
+	scaledObject *externalscaler.ScaledObjectRef,
```

Comment:
@ajanth97 can you remove this change? It doesn't appear necessary

Reply:
I added this to maintain consistency between the `StreamIsActive` and `IsActive` methods ? Don't you think that makes more sense to have the same variable name ?

Reply:
Yeah you are right. Actually I did try playing around with that method and then finally just reverted the changes in that method but I thought to leave the same variable name. I understand the value is not used but just for the sake of consistency.